lib/task.ml

```diff
@@ -22,7 +22,12 @@ let do_task f p =
     | TasksActive -> raise e
     | _ -> ()
 
-let setup_pool ~num_additional_domains =
+let named_pools = Hashtbl.create 8
+
+let setup_pool ?name ~num_additional_domains () =
+  if num_additional_domains < 0 then
+    raise (Invalid_argument "Task.setup_pool")
```

Comment:
You got for a more descriptive error message `Invalid argument "Task.setup_pool: num_additional_domains must be at least 0"`